adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/impl/AmqpContext.java

```diff
@@ -181,4 +183,25 @@ Sample getTimer() {
         return timer;
     }
 
+    /**
+     * Gets the value for the <em>sampling.priority</em> span tag to be used for OpenTracing spans created in connection
+     * with this AmqpContext.
+     *
+     * @return An <em>OptionalInt</em> containing the value for the <em>sampling.priority</em> span tag or an empty
+     *         <em>OptionalInt</em> if no priority should be set.
+     */
+    public OptionalInt getTraceSamplingPriority() {
+        return traceSamplingPriority;
+    }
+
+    /**
+     * Sets the value for the <em>sampling.priority</em> span tag to be used for OpenTracing spans created in connection
+     * with this AmqpContext.
+     *
+     * @param traceSamplingPriority The <em>OptionalInt</em> containing the <em>sampling.priority</em> span tag value or
+     *            an empty <em>OptionalInt</em> if no priority should be set.
+     */
+    public void setTraceSamplingPriority(final OptionalInt traceSamplingPriority) {
+        this.traceSamplingPriority = traceSamplingPriority;
```

Comment:
I think this will violate the contract described for the getter if `null` is passed in here ..

Reply:
Yes, I've added a null-check and javadoc.